lib/Test2/Compare/Negatable.pm

```diff
@@ -0,0 +1,119 @@
+package Test2::Compare::Negatable;
+use strict;
+use warnings;
+
+require overload;
+require Test2::Util::HashBase;
+
+sub import {
+    my ($pkg, $file, $line) = caller;
+
+    my $sub = eval <<"    EOT" or die $@;
+package $pkg;
+#line $line "$file"
+sub { overload->import('!' => 'clone_negate', fallback => 1); Test2::Util::HashBase->import('negate')}
```

Comment:
I don't understand why this is necessary. If `bool` returns a Test2::Compare::Bool object, the overload should be on the Test2::Compare::Bool class.